pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -292,12 +285,48 @@ func (s *NodeController) DoCheck(node *api.Node) []api.NodeCondition {
 			s.deletePods(node.Name)
 		}
 	}
-
 	conditions = append(conditions, *newReadyCondition)
 
+	// Check Condition: NodeSchedulable
+	oldSchedulableCondition := s.getCondition(node, api.NodeSchedulable)
+	newSchedulableCondition := s.checkNodeSchedulable(node)
```

Comment:
Our choice is to combine conditions in the scheduler, here:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/plugin/pkg/scheduler/factory/factory.go#L182

Or to combine them here. 

Given that I foresee handling eviction in the node controller, I agree that it makes sense to do it here.